### ISSUE TITLE
Custom Pills?

### ISSUE THREAD

**Issue by agluck91** [warpdotdev/Warp#5084]
### Dupe Check

- [X] I have [searched Warp feature requests](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3A%22Feature+requests%22) and there are no duplicates

### Describe the solution you'd like?

I would like the ability to add my own pills, or at least have more pill available.

I would love to see, tf, pulumi stack, node, and rust added to the list. Starship does this well. 

### How important is this feature to you?

3

### Warp Internal (ignore) - linear-label:39cc6478-1249-4ee7-950b-c428edfeecd1

None

**Comment by agluck91** [comment]
Just want to mention I love the pills combined with the SSH agent, simply awesome.

**Comment by dannyneira** [comment]
Thanks for this feature request! As a stopgap to this feature, Warp does support starship and other custom prompts listed here.

To anyone else interested in this feature, please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

**Comment by agluck91** [comment]
Thank you, I do have starship. It is not as good because it does not carry the prompt upstream to ssh client. You all knocked it out of the park. 

Logging into my K8s master and instantly seeing my context AS I am typing command. Instead of it always cluttering my shell. Wow!!

Can you all do something similar for pulumi, when I start typing my pulumi command I would see the stack I am targeting?

Would love to see what AWS profile I have set too. This would be really helpful. I change the profile by setting AWS_DEFAULT_PROFILE.  This should always be shown because some tools use the AWS config that is loaded. Like pulumi for instance and bash scripts ect 

node, python, rust should all be triggered by directory. For instance My node version is set per directory with Volta. You already have pyenv.

**Comment by agluck91** [comment]
I think I read that warp was thinking of going open source. Would definitely comtribute to this feature myself.

**Comment by dannyneira** [comment]
Thanks for this feature request @agluck91! We could open up an API for this feature to allow for custom chips in the future.

To anyone else interested in this feature, please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

**Comment by tukusejssirs** [comment]
I wish we could define custom chips using shell scripts, which could also define fg and bg colours of the output. For example, I want the `@` character between the username and hostname to be red when connected to a remote, otherwise it should be in normal/default colour.

> Moreover, I wish we could add literal text strings to the prompt, as some users might prefer different separators. For example, I want to add a static time in square brackets (e.g. `[ 4.19pm]`). Or I want to replace `git:(branch)` with `[branch]`, as I don’t use any other VCS.

**Comment by rafasoares** [comment]
>  As a stopgap to this feature, Warp does support starship and other custom prompts listed here.

Yes, but Warp chips look nicer by default and require less fiddling with shell scripts 😅 

In the meantime, @dannyneira, could/should we ask for specific chips? I'm assuming it's easier for the Warp to add new built-in chips than to design and implement a custom API.

**Comment by dannyneira** [comment]
@rafasoares yes extra chips make sense and maybe easier than a custom api/spec. I recommend making a [new feature request](https://github.com/warpdotdev/Warp/issues/new?template=02_feature_request.yml) for the new said chips with details. Note, there are already some existing specific chip requests out there, so make sure to search for them before opening new ones.

**Comment by wabiloo** [comment]
+1 for this feature. I make my own CLI tools which set contexts, and I'd love to be able to display those contexts with chips in the Warp prompt. 
I've done it with shell scripts for zsh, but like others, I find the Warp prompt much nicer!

**Comment by vexonius** [comment]
I would love to have this feature? Warp is now open sourced, right? Can we proceed?
